ui/components/app/auto-detect-token/auto-detect-token-modal.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useCallback, useContext } from 'react';
+
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Modal,
+  ModalContent,
+  ModalOverlay,
+  ModalHeader,
+  Box,
+  Text,
+  ButtonPrimary,
+  ButtonSecondary,
+  ButtonPrimarySize,
+  ButtonSecondarySize,
+} from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  AlignItems,
+  BorderRadius,
+  Display,
+  FlexDirection,
+  JustifyContent,
+  TextAlign,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { setUseTokenDetection } from '../../../store/actions';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+import { getProviderConfig } from '../../../ducks/metamask/metamask';
+import { getCurrentLocale } from '../../../ducks/locale/locale';
+import { ORIGIN_METAMASK } from '../../../../shared/constants/app';
+
+type AutoDetectTokenModalProps = {
+  isOpen: boolean;
+  onClose: (arg: boolean) => void;
+};
+function AutoDetectTokenModal({ isOpen, onClose }: AutoDetectTokenModalProps) {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const trackEvent = useContext(MetaMetricsContext);
+  const { chainId } = useSelector(getProviderConfig);
+
+  const handleTokenAutoDetection = useCallback(
+    (val) => {
+      trackEvent({
+        event: val
+          ? MetaMetricsEventName.TokenAutoDetectionEnableModal
+          : MetaMetricsEventName.TokenAutoDetectionDisableModal,
+        category: MetaMetricsEventCategory.Navigation,
+        properties: {
+          chain_id: chainId,
+          locale: getCurrentLocale,
+          referrer: ORIGIN_METAMASK,
+        },
+      });
+      dispatch(setUseTokenDetection(val));
+      onClose(val);
+    },
+    [dispatch],
+  );
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={() => onClose(true)}
+      isClosedOnOutsideClick={false}
+      isClosedOnEscapeKey={false}
+      className="mm-modal__custom-scrollbar mm-auto-detect-in-modal"
+      autoFocus={false}
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader
+          alignItems={AlignItems.center}
+          justifyContent={JustifyContent.center}
+        >
+          Enable token autodetection
+        </ModalHeader>
+        <Box
+          display={Display.Flex}
+          flexDirection={FlexDirection.Column}
+          paddingLeft={4}
+          paddingRight={4}
+        >
```

Comment:
Using `ModalBody` will help by managing overflow
```suggestion
        <ModalBody display={Display.Flex} flexDirection={FlexDirection.Column}>
```

<img width="389" alt="Screenshot 2024-05-28 at 8 46 13 AM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/0807fae1-044d-4a1d-94a1-776e7293d8a0">
<img width="430" alt="Screenshot 2024-05-28 at 8 46 40 AM" src="https://github.com/MetaMask/metamask-extension/assets/8112138/de43d059-9f25-46e0-a1fb-ce81d5ddcf27">